File: lw2/task4.pas
PROGRAM WorkWithQueryString(INPUT, OUTPUT);
USES
  DOS;  
FUNCTION GetQueryStringParameter(Param: STRING): STRING;
VAR
  QueryString: STRING;
BEGIN {GetQueryStringParameter}
  QueryString := GetEnv('QUERY_STRING');
  {Проверка вхождения в QueryString подстроки Param= или &Param=}
  IF (POS(Param + '=', QueryString) = 1) OR (POS('&' + Param + '=', QueryString) <> 0)
  THEN
    BEGIN
      QueryString := COPY(QueryString, POS(Param + '=', QueryString) + LENGTH(Param + '='), 255);
      IF POS('&', QueryString) <> 0
      THEN
        QueryString := COPY(QueryString, 1, POS('&', QueryString) - 1);
        GetQueryStringParameter := QueryString
    END
  ELSE      
    GetQueryStringParameter := ''    
END; {GetQueryStringParameter}
BEGIN {WorkWithQueryString}
  WRITELN('Content-Type: text/plain');
  WRITELN;     
  WRITELN('First Name: ', GetQueryStringParameter('first_name'));
  WRITELN('Last Name: ', GetQueryStringParameter('last_name'));
  WRITELN('Age: ', GetQueryStringParameter('age'))
END. {WorkWithQueryString}

File: lw3/task4.php
<?php
header("Content-Type: text/plain");

function getQueryStringParameter(string $name): ?string
{
    return isset($_GET[$name]) ? $_GET[$name] : null;
}

$firstName = getQueryStringParameter('first_name');
$lastName = getQueryStringParameter('last_name');
$age = getQueryStringParameter('age');
$email = getQueryStringParameter('email');
if ($email)
{
    $path = "./data/" . strtolower($email) . ".txt";
    if (file_exists($path))
    {
        $myFile = fopen($path, "r");
        getSubstringInFile(file($path));
        fclose($myFile);  
    }
    $myFile = fopen($path, "w");
    $somecontent = "First Name:" . $firstName . "\nLast Name:" . $lastName . "\nEmail:" . $email . "\nAge:" . $age;
    fwrite($myFile, $somecontent);
    fclose($myFile);
}
else
{
    echo 'have not email!'; 
}

function getSubstringInFile(array $arrayStrings): ?string
{ 
    global $firstName, $lastName, $age;
    $firstName = $firstName ?? trim(substr($arrayStrings[0], strpos($arrayStrings[0], ':') + 1));
    $lastName = $lastName ?? trim(substr($arrayStrings[1], strpos($arrayStrings[1], ':') + 1));
    $age = $age ?? trim(substr($arrayStrings[3], strpos($arrayStrings[3], ':') + 1)); 
    return '1';
}